eth/signer.go

```diff
@@ -132,24 +139,92 @@ func (s *Signer) CompactSign(buf []byte) ([]byte, error) {
 	return rsv[:64], nil
 }
 
-// SignWithNonce generates a 32-byte nonce with crypto/rand and returns
-// s.CompactSign(append(buf, nonce)). This can be verified in Solidity
-// via ecrecover with a hash value of
-// keccak256(abi.encodePacked(buf,nonce)).
-func (s *Signer) SignWithNonce(buf []byte) ([]byte, [32]byte, error) {
+// AppendRandomNonce appends random 32 bytes to the buffer, commonly used in
+// signature nonces.
+func AppendRandomNonce(buf []byte) ([]byte, [32]byte, error) {
 	var nonce [32]byte
 	if n, err := rand.Read(nonce[:]); n != 32 || err != nil {
 		return nil, nonce, fmt.Errorf("read 32 random bytes: got %d bytes with err %v", n, err)
 	}
 
-	sig, err := s.CompactSign(append(buf, nonce[:]...))
+	return append(buf, nonce[:]...), nonce, nil
+}
+
+// ToEthSignedMessageHash converts a given message to conform to the signed data
+// standard according to EIP-191.
+func ToEthPersonalSignedMessage(message []byte) []byte {
```

Comment:
I opted to expose these functions because I thought they might come in handy if there's a special task at hand for which we did not provide wrappers yet. So that people can access the building blocks and assemble them to their own needs. But maybe I overestimate this usecase.

Agree on the rename.  The only thing that I dislike about `eth.PersonalSignedMessage` is that it does not start with a verb. Maybe `eth.AddPersonalMessagePrefix`?

Reply:
I agree with your rationale of exposing for others when it comes to the personal-signed prefix (because it's a well-established standard) but not with the random nonce (because it's just something we made up). If we expose something then [people come to rely on it](https://www.hyrumslaw.com/) and we rope ourselves in.

Re the verb, since it doesn't modify the value but returns one, I don't think a verb is necessary when reading it at the call site; how about "With"? `x := eth.WithPersonalMessagePrefix(y)`

Reply:
You are totally right about overexposing. I made the random nonce internal now.
Also, I  agree on the rename.